src/models/metrics-table.config.ts

```diff
@@ -0,0 +1,369 @@
+        if (sort && sort.on === 'alphabetical') {
+            return Aggregation.OnEnum.Field;
+        } else if (sort && sort.on === 'count') {
+            return Aggregation.OnEnum.Count;
+        } else if (sort && sort.on === 'metric' && !!sort?.metric && sort.metric.metric !== 'count') {
+            return Aggregation.OnEnum.Result;
+        } else {
+            return Aggregation.OnEnum.Count;
+        }
+    }
+
+    /** Returns the Aggregation.OrderEnum to apply to arlas aggregation request object. */
+    private getSortOrder(sort: MetricsTableSortConfig): Aggregation.OrderEnum {
+        if (sort.order === 'asc') {
+            return Aggregation.OrderEnum.Asc;
+        } else if (sort.order === 'desc') {
+            return Aggregation.OrderEnum.Desc;
+        } else {
+            /** sort.order should always be specified. This block is for code safety. */
+            return Aggregation.OrderEnum.Desc;
+        }
+    }
+
+    public mergeResponses(baseResponse: AggregationResponse, complementaryResponse: AggregationResponse): AggregationResponse {
+        const mergedResponse = Object.assign({}, baseResponse);
+
+        const baseArray = this.getComparableArray(baseResponse);
+        const complementaryArray = this.getComparableArray(complementaryResponse);
+        const baseElements = baseResponse.elements;
+        const complementarElements = complementaryResponse.elements;
+        const mergedElements = [];
+        let i = 0, j = 0;
+        const compare = (a: number, b: number): boolean => this.getSortOrder(this.sort) === Aggregation.OrderEnum.Asc ? a < b : a > b;
+        // Merge arrays until one is exhausted
+        while (i < baseElements.length && j < complementarElements.length) {
+            if (compare(baseArray[i], complementaryArray[j])) {
+                mergedElements.push(baseElements[i]);
+                i++;
+            } else {
+                mergedElements.push(complementarElements[j]);
+                j++;
+            }
+        }
+
+        // Add remaining elements from array1, if any
+        while (i < baseElements.length) {
+            mergedElements.push(baseElements[i]);
+            i++;
+        }
+
+        // Add remaining elements from array2, if any
+        while (j < complementarElements.length) {
+            mergedElements.push(complementarElements[j]);
+            j++;
+        }
+        mergedResponse.elements = mergedElements;
+        return mergedResponse;
+    }
+
+    private getComparableArray(response: AggregationResponse): number[] {
+        if (this.isSortOnCount()) {
+            return response.elements.map(e => e.count);
+        } else {
+            const metricConfig = this.sort.metric;
+            return response.elements.map(e => e.metrics?.find(m => (m.field === metricConfig.field.replace(/\./g, '_')
+             && m.type === metricConfig.metric))?.value);
+        }
+    }
+
+    private isSortOnCount() {
```

Comment:
In the first two conditions of the OR, it does not correspond to the name of the method of isSortOnCount, is it because of the default being a sort on count ?

Reply:
totally

Reply:
Maybe that could be added as a small comment to make it clearer for future readers ?

Reply:
added